crates/attestation/src/crypto/crl.rs

```diff
@@ -57,18 +49,18 @@ impl<'a> PkiPathCRLCheck<'a> for PkiPath<'a> {
                 break;
             }
             let cert = cert.unwrap();
-            let next = next.unwrap();
+            let next = *next.unwrap();
 
             let crls = {
                 let mut crls = vec![];
-                if let Some(crl) = pairs.get_crl_by_issuer(&cert.tbs_certificate.subject) {
-                    crls.push(crl);
-                } else {
-                    let urls = cert.tbs_certificate.get_crl_urls()?;
-                    for url in urls {
-                        if let Some(crl) = pairs.get_crl_by_url(&url) {
-                            crls.push(crl);
-                        }
+                let mut urls = next.tbs_certificate.get_crl_urls()?;
+                // Some end certs use the parent cert CRL
+                if urls.is_empty() {
+                    urls = cert.tbs_certificate.get_crl_urls()?;
+                }
+                for url in urls {
+                    if let Some(crl) = pairs.get_crl_by_url(&url) {
+                        crls.push((url, crl));
```

Comment:
We should take the first one that we can find.
If there's multiple distribution points, those are supposed to be multiple ways to get the same CRL (other than if the CRLs are segmented by reason code, but that is recommended against, and I've never seen this happen, and I'm also not seeing it in the SGX or SNP certs in the repo).

Reply:
Done